src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,217 @@
+            log.info(f"Zephyr test was skipped.")
+            result = TestResult.SKIPPED
+        elif error is not None:
+            log.error(f"Zephyr test failed with error: {error.text}")
+            result = TestResult.ERROR
+        else:
+            log.info(f"Zephyr test PASSED.")
+            result = TestResult.PASSED
+        return result
+
+    def wait_test(self) -> TestResult:
+        """Wait for the test to finish
+
+        :return: The result of the test, taken from the xunit output of twister
+        """
+        if self.process is None:
+            raise ZephyrError("Twister was not started so can not wait.")
+
+        # Read the twister console for log output
+        while True:
+            line = self._readline()
+            if line is None:
+                break
+
+            # Look for relevant messages in the output
+            if "OUTPUT:  PASS - " in line:
+                log.debug(f"Zephyr test PASSED.")
+            elif "OUTPUT:  FAIL - " in line:
+                log.debug(f"Zephyr test FAILED.")
+
+        # Wait for the twister process to exit
+        ret = self.process.wait()
+
+        # Get the result from the xunit file
+        result = self._parse_xunit(str(Path(self.outdir) / "twister_report.xml"))
+
+        self.process = None
+        return result
+
+
+class ZephyrTestAuxiliary(SimpleAuxiliaryInterface):
+    """Auxiliary for Zephyr test interaction using the twister commandline tool
+
+    The functionality includes test case execution and result collection.
+
+    """
+
+    def __init__(
+        self,
+        twister_path: str = "twister",
+        test_directory: Optional[str] = None,
+        test_name: Optional[str] = None,
+        wait_for_start: bool = True,
+        **kwargs,
+    ) -> None:
+        """Initialize the auxiliary
+
+        :param twister_path: Path to the twister tool
+        :param test_directory: The directory to search for the Zephyr test project
+        :param testcase_name: The name of the Zephyr test
+        :param wait_for_start: Wait for Zyephyr test start
+
+        """
+        self.twister_path = twister_path
+        self.test_directory = test_directory
+        self.test_name = test_name
+        self.wait_for_start = wait_for_start
+        self.twister = Twister(twister_path)
+        super().__init__(**kwargs)
+
+    def start_test(
+        self, test_directory: Optional[str] = None, test_name: Optional[str] = None
+    ) -> None:
+        """Start the Zephyr test
+
+        :param test_directory: The directory to search for the Zephyr test project. Defaults to the test_directory from YAML.
+        :param testcase_name: The name of the Zephyr test. Defaults to the testcase_name from YAML.
+        """
+        test_directory = (
+            test_directory if test_directory is not None else self.test_directory
+        )
+        test_name = test_name if test_name is not None else self.test_name
+        if test_directory is None:
```

Comment:
But then you would be forced to define these parameters in the yaml, even if it does not fit your usecase.(You might want to execute multiple zephyr testcases)
Then it would be better to completely remove these parameters from yaml. Then you always have to specificy the testcase in the start_test call.